install.sh

```diff
@@ -90,106 +87,121 @@ function curl () {
+      fi
+    done
   fi
-done
+else
+  sudo mkdir ${CREW_PREFIX}
+fi
+
+# This will allow things to work without sudo.
 sudo chown "$(id -u)":"$(id -g)" "${CREW_PREFIX}"
+# This will create the directories.
+crew_folders="bin cache doc docbook include lib/crew/packages lib$LIB_SUFFIX libexec man sbin share var etc/crew/meta etc/env.d tmp/crew/dest"
+# shellcheck disable=SC2086
+# Quoting crew_folders leads to breakage
+(cd "${CREW_PREFIX}"; mkdir -p ${crew_folders})
 
-# Delete ${CREW_PREFIX}/{var,local} symlinks on some Chromium OS distro
-# if they exist.
-[ -L ${CREW_PREFIX}/var ] && sudo rm -f "${CREW_PREFIX}/var"
-[ -L ${CREW_PREFIX}/local ] && sudo rm -f "${CREW_PREFIX}/local"
 
-# Prepare directories.
-for dir in "${CREW_CONFIG_PATH}/meta" "${CREW_DEST_DIR}" "${CREW_PACKAGES_PATH}" "${CREW_CACHE_DIR}" ; do
-  if [ ! -d "${dir}" ]; then
-    mkdir -p "${dir}"
-  fi
-done
+# Remove old git config directories if they exist.
+find "${CREW_LIB_PATH}" -mindepth 1 -delete
 
-echo_info "\nDownloading information for Bootstrap packages..."
-echo -en "${GRAY}"
-# Use parallel mode if available.
-CURL_OPTS="-OL"
-[[ "$(curl --help curl)" =~ --parallel ]] && CURL_OPTS+="Z"
-BOOTSTRAP_PKGS="${BOOTSTRAP_PACKAGES// /.rb,}.rb"
-cd "${CREW_PACKAGES_PATH}" && curl "${CURL_OPTS}" "${URL}/packages/{${BOOTSTRAP_PKGS}}"
-echo -e "${RESET}"
+# Download the chromebrew repository.
+curl -L -O --progress-bar https://github.com/"${OWNER}"/"${REPO}"/tarball/"${BRANCH}"
+tar -xzf "${BRANCH}" --strip-components=1 --directory "${CREW_LIB_PATH}"
 
-# Prepare url and sha256.
+# Prepare urls and sha256s variables.
 urls=()
 sha256s=()
 
-case "${ARCH}" in
-"armv7l"|"aarch64")
-  if ! type "xz" > /dev/null; then
-    urls+=('https://github.com/snailium/chrome-cross/releases/download/v1.8.1/xz-5.2.3-chromeos-armv7l.tar.gz')
-    sha256s+=('4dc9f086ee7613ab0145ec0ed5ac804c80c620c92f515cb62bae8d3c508cbfe7')
+BOOTSTRAP_PACKAGES="zstd crew_mvdir ruby git ca_certificates"
+
+# i686 requires gcc and openssl.
+[ "${ARCH}" == "i686" ] && BOOTSTRAP_PACKAGES+=" gcc_lib openssl"
+# Get the URLs and hashes of the bootstrap packages.
+for package in $BOOTSTRAP_PACKAGES; do
+  cd "${CREW_LIB_PATH}/packages"
+  [[ "$(sed -n '/binary_sha256/,/}/p' "${package}.rb")" =~ .*${ARCH}:[[:blank:]]*[\'\"]([^\'\"]*) ]]
+    sha256s+=("${BASH_REMATCH[1]}")
+
+  name=$(sed -n '3s/class\(.*\)<.*/\L\1/;3s/^[ \t]*//;3s/[ \t]*$//p' "${package}.rb")
+  version=$(sed -n '6s/version \(.*\)/\1/;6s/^[ \t]*//;6s/[ \t]*$//;6s/ *#.*//p' "${package}.rb" | tr -d "'")
+
+  # This is really ugly, FIXME after #7082 is merged.
+  if [[ "${package}" == "zstd" ]]; then
+    if [[ "$ARCH" == "aarch64" ]]; then
+      urls+=("https://gitlab.com/api/v4/projects/26210301/packages/generic/${name}/${version}_armv7l/${name}-${version}-chromeos-armv7l.tar.xz")
+    else
+      urls+=("https://gitlab.com/api/v4/projects/26210301/packages/generic/${name}/${version}_${ARCH}/${name}-${version}-chromeos-${ARCH}.tar.xz")
```

Comment:
Make a variable for `https://gitlab.com/api/v4/projects/26210301/packages/generic/` to shorten the url?